hugo/assets/scripts/domainmodel/d3tree.tsx

```diff
@@ -22,22 +22,22 @@ export default function D3Tree({ data }: TreeProps) {
   useEffect(() => {
     if (!svgRef.current) return;
 
-    // get the size of the tree
-    const getChildSize = (child: TreeNode): number => {
+    // get the leaf nodes of the tree
+    const getLeafNodes = (child: TreeNode): number  => {
       if (!child.children) return 1;
-      let amount = child.children.length;
-
-      // fastest way to iterate over an array
-      const length = child.children.length;
-      for (let i = 0; i < length; i++) {
-        amount += getChildSize(child.children[i]);
-      }
-      return amount;
+      if(child.children.length === 0) return 1;
+      return child.children.map(getLeafNodes).reduce((a, b) => a + b);
     };
 
-    const size = getChildSize(data);
-    const height = size * 20;
-    const width = size * 18;
+    // get the longest path in the tree
+    const getLongestPath = (child: TreeNode): number => {
+      if (!child.children) return 1;
+      if(child.children.length === 0) return 1;
+      return 1 + Math.max(...child.children.map(getLongestPath));
+    };
+
+    const height = getLeafNodes(data) * 60;
```

Comment:
A minor thing: You use magic numbers. Maybe give them a name. Especially when one number relates to the amount of another number. I guess these numbers relate to the max label size of the nodes or?
If there is no relation between these numbers and they are unique as they are used here, I think you can skip this comment of mine.

It is about code quality. Those hidden relationships can cost time for the next guy working on this file.